pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/consumer/PulsarKafkaConsumer.java

```diff
@@ -311,6 +325,9 @@ public void unsubscribe() {
                 long offset = MessageIdUtils.getOffset(msgId);
 
                 TopicPartition tp = new TopicPartition(topic, partition);
+                if (lastReceivedOffset.get(tp) == null) {
```

Comment:
can you explain why you are adding this logic here?

Reply:
Well, I noticed that lastReceivedOffset is cleared whenever a seek-to-end or seek-to-front is called.  So looking into it, I found that whenever lastReceivedOffset.get(tp) returns null, it would indicate the current offset position is invalid (or could not be found since we are using a map). Taking advantage of this fact, I would just call seekToEnd or seekToFront to achieve the same behavior as KafkaConsumer. Because in KafkaConsumer internals, their position() method and seekToEnd() and seekToStart() uses the same approach.